src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockMigrationService.cs

```diff
@@ -1,40 +1,456 @@
+                                        plan.Type,
+                                        plan.Strategy,
+                                        map.BoundaryLevel,
+                                        boundaryLevels.Single(level => level.Id == map.BoundaryLevel).Level);
+                                }
+                                case SetBoundaryLevelToMatchSingleNonCountryDataSetLevel:
+                                {
+                                    var nonCountryGeographicLevel = plan
+                                        .DataSetGeographicLevels
+                                        .Single(geographicLevel => geographicLevel != GeographicLevel.Country);
+
+                                    var createdDate = await GetDataBlockCreatedDate(dataBlock);
+
+                                    var boundaryLevel = GetLatestBoundaryLevelAtCreationTime(
+                                        createdDate,
+                                        nonCountryGeographicLevel,
+                                        boundaryLevels);
+
+                                    dataBlock.Query.BoundaryLevel = boundaryLevel.Id;
+                                    map.BoundaryLevel = boundaryLevel.Id;
+                                    await SaveChanges(dataBlock, dryRun);
+                                    return new MapMigrationResult(
+                                        plan.DataBlockId,
+                                        plan.Type,
+                                        plan.Strategy,
+                                        map.BoundaryLevel,
+                                        boundaryLevels.Single(level => level.Id == map.BoundaryLevel).Level);
+                                }
+                                case QuestionWhichCorrectGeoLevelToSelect:
+                                    return new MapMigrationResult(
+                                        plan.DataBlockId,
+                                        plan.Type,
+                                        plan.Strategy,
+                                        null,
+                                        null);
+                                case FixMapConfigNoDataSetGeographicLevels:
+                                    return new MapMigrationResult(
+                                        plan.DataBlockId,
+                                        plan.Type,
+                                        plan.Strategy,
+                                        null,
+                                        null);
+                                case FixMapConfigInvalidDataSetGeographicLevels:
+                                    return new MapMigrationResult(
+                                        plan.DataBlockId,
+                                        plan.Type,
+                                        plan.Strategy,
+                                        null,
+                                        null);
```

Comment:
I think you could stack these three cases and just have one return for all three? Realise this is getting a bit pedantic for migration code :)

Reply:
Hey no, it's totally valid and a 1 second change - done!

For throwaway migration code, if it was something more involved then I might push back, but this is all good!